packages/react-core/src/core/providers/thirdweb-wallet-provider.tsx

```diff
@@ -353,5 +408,35 @@ export function ThirdwebWalletProvider(
 }
 
 export function useThirdwebWallet() {
-  return useContext(ThirdwebWalletContext);
+  const ctx = useContext(ThirdwebWalletContext);
+  if (!ctx) {
+    throw new Error(
+      `useThirdwebWallet() can only be used inside <ThirdwebProvider />`,
+    );
+  }
+  return ctx;
+}
+
+async function getLastConnectedWalletInfo() {
+  const str = await lastConnectedWalletStorage.getItem(
+    LAST_CONNECTED_WALLET_STORAGE_KEY,
+  );
+  if (!str) {
+    return null;
+  }
+
+  return JSON.parse(str) as LastConnectedWalletInfo;
+}
+
+async function saveLastConnectedWalletInfo(
+  walletInfo: LastConnectedWalletInfo,
+) {
+  try {
+    await lastConnectedWalletStorage.setItem(
+      LAST_CONNECTED_WALLET_STORAGE_KEY,
+      JSON.stringify(walletInfo),
+    );
+  } catch (e) {
+    console.error(e);
```

Comment:
can we add a description to this log? Something like: `console.error("Error saving the last connected wallet info", e)`